modules/intensity_transform/src/BIMEF.cpp

```diff
@@ -0,0 +1,575 @@
+
+    Mat_<float> Y_mat(static_cast<int>(Y_vec.size()), 1, Y_vec.data());
+    float opt_k = static_cast<float>(minimize_scalar_bounded(Y_mat, 1, 7));
+
+    return applyK(I, opt_k, a, b, -0.01f);
+}
+
+void BIMEF_impl(InputArray input_, OutputArray output_, float mu, float *k, float a, float b)
+{
+    CV_INSTRUMENT_REGION()
+
+    Mat input = input_.getMat();
+    if (input.empty())
+    {
+        return;
+    }
+    CV_CheckTypeEQ(input.type(), CV_8UC3, "Input image must be 8-bits color image (CV_8UC3).");
+
+    Mat_<Vec3f> imgDouble;
+    input.convertTo(imgDouble, CV_32F, 1/255.0);
+
+    // t: scene illumination map
+    Mat_<float> t_b(imgDouble.size());
+    t_b.forEach(
+        [&](float &pixel, const int * position) -> void
+        {
+            pixel = std::max(std::max(imgDouble(position[0], position[1])[0],
+                                      imgDouble(position[0], position[1])[1]),
+                            imgDouble(position[0], position[1])[2]);
+        }
+    );
+
+    const float lambda = 0.5;
+    const float sigma = 5;
+
+    Mat_<float> t_b_resize;
+    resize(t_b, t_b_resize, Size(), 0.5, 0.5);
+
+    Mat_<float> t_our = tsmooth(t_b_resize, lambda, sigma);
+    resize(t_our, t_our, t_b.size());
+
+    // k: exposure ratio
+    Mat_<Vec3f> J;
+    if (k == NULL)
+    {
+        Mat_<uchar> isBad(t_our.size());
+        isBad.forEach(
+            [&](uchar &pixel, const int * position) -> void
+            {
+                pixel = t_our(position[0], position[1]) < 0.5 ? 1 : 0;
+            }
+        );
+
+        J = maxEntropyEnhance(imgDouble, isBad, a, b);
+    }
+    else
+    {
+        J = applyK(imgDouble, *k, a, b);
+
+        // fix overflow
+        J.forEach(
+            [](Vec3f &pixel, const int * /*position*/) -> void
+            {
+                pixel(0) = std::min(1.0f, pixel(0));
+                pixel(1) = std::min(1.0f, pixel(1));
+                pixel(2) = std::min(1.0f, pixel(2));
+            }
+        );
+    }
+
+    // W: Weight Matrix
+    Mat_<float> W(t_our.size());
+    pow(t_our, mu, W);
+
+
+    output_.create(input.size(), CV_8UC3);
+    Mat output = output_.getMat();
+    output.forEach<Vec3b>(
+        [&](Vec3b &pixel, const int * position) -> void
+        {
+            float w = W(position[0], position[1]);
+            pixel(0) = saturate_cast<uchar>((imgDouble(position[0], position[1])[0] * w + J(position[0], position[1])[0] * (1 - w)) * 255);
+            pixel(1) = saturate_cast<uchar>((imgDouble(position[0], position[1])[1] * w + J(position[0], position[1])[1] * (1 - w)) * 255);
+            pixel(2) = saturate_cast<uchar>((imgDouble(position[0], position[1])[2] * w + J(position[0], position[1])[2] * (1 - w)) * 255);
+        }
+    );
+}
+#else
+void BIMEF_impl(InputArray, OutputArray, float, float *, float, float)
+{
+    CV_Error(Error::StsNotImplemented, "This algorithm requires OpenCV built with the Eigen library.");
+}
+#endif
+} //anonymous namespace
```

Comment:
Please fix inconsistent `namespace` and `#ifdef` (`cmake . -DWITH_EIGEN=OFF` builds are failed) to resolve failed builds.

Instead of anonymous namespace you may add "static" modifier for `BIMEF_impl` stub.

Reply:
Fix with `static`.